sky/utils/log_utils.py

```diff
@@ -196,21 +196,33 @@ def readable_time_duration(start: Optional[float],
 
 
 def create_and_symlink_log_dir(log_dir: str):
-    """Creates a new log directory and symlinks to it from parent directory.
+    """Create a log directory and symlink to it from parent directory.
+
+    If file operations fail, will log a warning and return without error.
 
     Args:
         log_dir (str): Expanded log directory path.
     """
-    if log_dir == os.devnull:
+    try:
+        os.makedirs(log_dir, exist_ok=True)
+    except OSError:
+        logger.warning(f'Failed to create log directory at {log_dir!r}.'
+                       'Check if there is a file with the same name.')
```

Comment:
Perhaps we should not wrap this in try except? This is a fairly critical operation, if this fails we should surface to the user (same behavior as before this PR).

Reply:
Yeah I think that makes sense. The `/dev/null` case will only pass in `/dev` to this function, so an error would be unusual. Will revert to that logic.